pkg/assets/create/creater.go

```diff
@@ -0,0 +1,177 @@
+package create
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+
+	"github.com/openshift/library-go/pkg/assets"
+)
+
+type CreateOptions struct {
+	// Filters allows to filter which files we will read from disk.
+	Filters []assets.FileInfoPredicate
+
+	// Context specifies time to terminate creation and report last error
+	Context context.Context
+
+	client dynamic.Interface
+	mapper meta.RESTMapper
+}
+
+// EnsureManifestsCreated ensures that all resource manifests from the specified directory are created.
+// This function will keep retrying creation until no errors are reported. After specified timeout, the last set
+// of observed errors are returned to caller.
+func EnsureManifestsCreated(manifestDir string, restConfig *rest.Config, options CreateOptions) error {
+	var err error
+
+	options.client, err = dynamic.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	dc := discovery.NewDiscoveryClientForConfigOrDie(restConfig)
+	gr, err := restmapper.GetAPIGroupResources(dc)
+	if err != nil {
+		return err
+	}
+	options.mapper = restmapper.NewDiscoveryRESTMapper(gr)
```

Comment:
how ugly to use that as state. Just pass the mapper down.

Reply:
Not that this mapper does not update automatically in case of the CRDs.

Reply:
@sttts @soltysh added logic to handle this case. If I get rest mapping error (no mapping), i will reload the discovery to get it updated...